Sources/Testing/Testing.docc/exit-testing.md

```diff
@@ -0,0 +1,155 @@
+- If the body will cause the child process to exit with some failure, but the
+  exact status reported by the system is not important, pass
+  ``ExitTest/Condition/failure``.
+- If you need to check for a specific exit code or signal, pass
+  ``ExitTest/Condition/exitCode(_:)`` or ``ExitTest/Condition/signal(_:)``.
+
+When the child process exits, the parent process resumes and compares the exit
+status of the child process against the expected exit condition you passed. If
+they match, the exit test has passed; otherwise, it has failed and the testing
+library records an issue.
+
+### Gather output from the child process
+
+By default, the child process is configured without a standard output or
+standard error stream. If your test needs to review the content of either of
+these streams, you can pass its key path to ``expect(exitsWith:observing:_:sourceLocation:performing:)``
+or ``require(exitsWith:observing:_:sourceLocation:performing:)``:
+
+```swift
+extension Customer {
+  func eat(_ food: consuming some Food) {
+    print("Let's see if I want to eat \(food)...")
+    precondition(food.isDelicious, "Tasty food only!")
+    precondition(food.isNutritious, "Healthy food only!")
+    ...
+  }
+}
+
+@Test func `Customer won't eat food unless it's delicious`() async {
+  let result = await #expect(
+    exitsWith: .failure,
+    observing: [\.standardOutputContent]
+  ) {
+    var food = ...
+    food.isDelicious = false
+    Customer.current.eat(food)
+  }
+  if let result {
+    #expect(result.standardOutputContent.contains(UInt8(ascii: "L")))
+  }
+}
+```
+
+- Note: The content of the standard output and standard error streams may
+  contain any arbitrary sequence of bytes, including sequences that are not
+  valid UTF-8 and cannot be decoded by [`String.init(cString:)`](https://developer.apple.com/documentation/swift/string/init(cstring:)-6kr8s).
+  These streams are globally accessible within the child process, and any code
+  running in an exit test may write to it including the operating system and any
+  third-party dependencies you have declared in your package.
+
+The testing library always sets ``ExitTest/Result/statusAtExit`` to the actual
+exit status of the child process (as reported by the system) even if you do not
+pass it.
+
+### Constraints on exit tests
+
+#### State cannot be captured
+
+Exit tests cannot capture any state originating in the parent process or from
+the enclosing lexical context. For example, the following exit test will fail to
+compile because it captures a variable declared outside the exit test itself:
+
+```swift
+@Test func `Customer won't eat food unless it's nutritious`() async {
+  let isNutritious = false
+  await #expect(exitsWith: .failure) {
+    var food = ...
+    food.isNutritious = isNutritious // ❌ ERROR: trying to capture state here
+    Customer.current.eat(food)
+  }
+}
+```
+
+#### Exit tests cannot be nested
+
+An exit test cannot run within another exit test.
```

Comment:
```suggestion
An exit test can't run another exit test.
```

It means the same to say "a test can't run a child test" as "a test can't run in a parent test", and it's simpler.